### DISCUSSION TITLE
How to set allowed Destinations using regex from name of cluster instead of server

### DISCUSSION THREAD

**Discussion by emanuelflp** [argoproj/argo-cd#7063] [Q&A]
Hello to all.

Can we set the parameter `Allowed Destinations` as a regex from name parameter in AppProject resource?

My situation is:
We want to create 4 projects, one for each environment that we have(shr, hml, dev, prd). We have an automation that auto register new clusters in an existent ArgoCD and this automation even register the cluster name as ENV_NAME-CLUSTER_NAME. I want to split our applications by environment/projects, but we need a way to "add" the cluster to a project automatically.

E.g.: cluster with name `shr-cluster1` can receive applications from project `shr` as this project has a filter `shr-*` in allowed destinations.

**Comment by deepali-personal** [comment]
> Hello to all.
> 
> Can we set the parameter `Allowed Destinations` as a regex from name parameter in AppProject resource?
> 
> My situation is:
> We want to create 4 projects, one for each environment that we have(shr, hml, dev, prd). We have an automation that auto register new clusters in an existent ArgoCD and this automation even register the cluster name as ENV_NAME-CLUSTER_NAME. I want to split our applications by environment/projects, but we need a way to "add" the cluster to a project automatically.
> 
> E.g.: cluster with name `shr-cluster1` can receive applications from project `shr` as this project has a filter `shr-*` in allowed destinations.

Hi @emanuelflp ,

Could you please provide your steps to auto register new clusters in an existent ArgoCD and how the automation registers the cluster name as ENV_NAME-CLUSTER_NAME?

**Comment by todaywasawesome** [comment]
You might want to check out Argo CD Autopilot which does this. [Argo CD Autopilot](https://github.com/argoproj-labs/argocd-autopilot)
